### PR TITLE
Item Spawning and Collision

### PR DESCRIPTION
**Pull request by OmKurkure**
1. Added Item Spawning Functionality according to CSV data
2. Added playeritemCollision. Player successfully picks up item, however there are some still things to brainstorm about:
-> The GetMetaData() function in CollisionController is specific player item collisions, just wanted to test it out
-> Items still spawn in the room after picking up and revisiting later
3. Refactored Projectile classes to support collision checking
4. Removed hitbox drawing

### CODE REVIEW

File: MonoZelda/Commands/CollisionCommands/PlayerItemCollisionCommand.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            collidableDict.Enabled = false;
+            collisionController.RemoveCollidable(collidableA);
+        }
+        else
+        {
+            SpriteDict collidableDict = collidableB.CollidableDict;
+            collidableDict.Enabled = false;
+            collisionController.RemoveCollidable(collidableB);
+        }
+    }
```
Comment (RobertEGreenslade12) [inline MonoZelda/Commands/CollisionCommands/PlayerItemCollisionCommand.cs:39]: So, does this code just destroy the item? It doesn't actually equip the player with the item or anything?

File: MonoZelda/Commands/GameCommands/PlayerMoveCommand.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ public PlayerMoveCommand(Player player)
 
     public void Execute(params object[] metadata)
     {
-        Keys pressedKey = (Keys) metadata[0];
+        Keys pressedKey = (Keys)metadata[0];
```
Comment (RobertEGreenslade12) [inline MonoZelda/Commands/GameCommands/PlayerMoveCommand.cs:25]: Why did you remove the space here? Is that standard c# notation? I'm just curious I have no clue.

File: MonoZelda/Controllers/CollisionController.cs [hunk 3]
```diff
@@ -91,7 +91,14 @@ private void HandleCollision(Collidable collidableA, Collidable collidableB, par
     private object[] GetMetadata(Collidable collidableA, Collidable collidableB)
     {
         // We need to implement this method to give back information about the collision
-        return null;
+        var metadata = new object[]
+        {
+            collidableA,
+            collidableB,
+            this,
+        };
```
Comment (RobertEGreenslade12) [inline MonoZelda/Controllers/CollisionController.cs:99]: I know this is all temporary, but I am voicing that I don't think we should pass the collision controller to commands so we can keep coupling as loose as possible. I do understand that we can make an exception if it is needed.

File: MonoZelda/Items/ItemFactory.cs [hunk 4]
```diff
@@ -1,27 +1,31 @@
...
-    public T CreateItem<T>() where T : IItem
+    public void CreateItem(ItemList itemName, Point spawnPosition)
     {
-        // Create an instance of the item, passing in the common objects to the constructor
-        return (T)Activator.CreateInstance(typeof(T), collisionController, graphicsDevice);
+        var itemDict = new SpriteDict(contentManager.Load<Texture2D>("Sprites/items"), SpriteCSVData.Items, 0, new Point(0, 0));
+        var itemType = Type.GetType($"MonoZelda.Items.ItemClasses.{itemName}");
+        IItem item = (IItem)Activator.CreateInstance(itemType);
+        item.itemSpawn(itemDict, spawnPosition, collisionController);
+
```
Comment (RobertEGreenslade12) [inline MonoZelda/Items/ItemFactory.cs:28]: Is this code what connects the item factory to the level loading? Will items now spawn for each room when this PR is merged?

File: MonoZelda/Scenes/DungeonScene.cs [hunk 5]
```diff
@@ -41,15 +41,15 @@ public void LoadContent(ContentManager contentManager)
 
         //create player and player collision
         player = new Player();
-        Collidable playerHitbox = new Collidable(new Rectangle(100, 100, 50, 50), graphicsDevice, CollidableType.Player);
+        Collidable playerHitbox = new Collidable(new Rectangle(100, 100, 50, 50), CollidableType.Player);
```
Comment (RobertEGreenslade12) [inline MonoZelda/Scenes/DungeonScene.cs:44]: Should the player hitbox be created within player so it can be removed from game? Doesn't need to happen in this PR. Maybe make a ticket if you think it's something that should be done.

File: MonoZelda/Scenes/DungeonScene.cs [hunk 6]
```diff
@@ -61,30 +61,25 @@ public void LoadContent(ContentManager contentManager)
...
     }
 
 
+
     private void LoadRoom(ContentManager contentManager)
     {
         LoadRoomTextures(contentManager);
         CreateStaticColliders();
-        SpawnItems();
+        SpawnItems(contentManager);
```
Comment (RobertEGreenslade12) [inline MonoZelda/Scenes/DungeonScene.cs:72]: This must be what spawns the items when a new room loads.

**Review by RobertEGreenslade12 — APPROVED** [review]

**Comment by RobertEGreenslade12** [comment]
I approved, the comments I made were all just thoughts I had along the way. I haven't tested, but read through the code extensively. I am not near a device to test.

File: MonoZelda/Commands/CollisionCommands/PlayerItemCollisionCommand.cs (hunk 1 above)
Comment (OmKurkure) [reply MonoZelda/Commands/CollisionCommands/PlayerItemCollisionCommand.cs:39]: Yeah, I am not sure, but I think after the item is destroyed, we would have to update Link's inventory, which would like a inventory class, and update that link has picked up the item in the GUI of the game.

File: MonoZelda/Items/ItemFactory.cs (hunk 4 above)
Comment (OmKurkure) [reply MonoZelda/Items/ItemFactory.cs:28]: Yes, this method is called in the DungeonScene class, where the items from the ItemSpawn.GetItemSpawn() list are iterated over and createItem is called on each of them. Updating the Google sheet will items will automically spawn items because of how room loading has been implemented

File: MonoZelda/Scenes/DungeonScene.cs (hunk 6 above)
Comment (OmKurkure) [reply MonoZelda/Scenes/DungeonScene.cs:72]: Correct

File: MonoZelda/Controllers/CollisionController.cs (hunk 3 above)
Comment (OmKurkure) [reply MonoZelda/Controllers/CollisionController.cs:99]: I agree, we would have to implemented some other way to remove collidables from the collisionController list, rather than passing the collisionController object to multiple classes

File: MonoZelda/Scenes/DungeonScene.cs (hunk 5 above)
Comment (OmKurkure) [reply MonoZelda/Scenes/DungeonScene.cs:44]: Yes, creating player hitbox insided the player class would be ideal.

**Comment by xsedim** [comment]
Collision hitbox drawing should not be removed— it was my plan to add a key to toggle hitboxes for future debugging. I don't think the PR should be merged until this is changed as adding the feature back in later would be more work in my opinion than fixing it here. 

Other than that I tested the branch on my device and everything works.
